jujugui/static/gui/src/app/store/env/sandbox.js

```diff
@@ -1346,15 +1353,99 @@ YUI.add('juju-env-sandbox', function(Y) {
     },
 
     /**
+      Handle Cloud.UpdateCredentials messages.
+
+      @method handleCloudUpdateCredentials
+      @param {Object} data The contents of the API arguments.
+      @param {Object} client The active ClientConnection.
+      @param {Object} state An instance of FakeBackend.
+    */
+    handleCloudUpdateCredentials: function(data, client, state) {
+      const entries = data.params.credentials;
+      if (!entries.length) {
+        client.receive({
+          'request-id': data['request-id'],
+          response: {results: []}
+        });
+        return;
+      }
+      const cloudCredentialsStorage = this._getCredentialsStorage(state);
+      const results = entries.map(entry => {
+        if (!entry.tag.startsWith('cloudcred-')) {
+          return {error: {message: entry.tag + ' is not a valid tag'}};
+        }
+        cloudCredentialsStorage[entry.tag] = entry.credential;
+        return {};
+      });
+      client.receive({
+        'request-id': data['request-id'],
+        response: {results: results}
+      });
+    },
+
+    /**
+      Handle Cloud.RevokeCredentials messages.
+
+      @method handleCloudRevokeCredentials
+      @param {Object} data The contents of the API arguments.
+      @param {Object} client The active ClientConnection.
+      @param {Object} state An instance of FakeBackend.
+    */
+    handleCloudRevokeCredentials: function(data, client, state) {
+      const entities = data.params.entities;
```

Comment:
As mentioned, in this case this is safe, as we are just parsing the request prepared in controllerAPI.js.